rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1152,6 +1156,64 @@ rmw_context_impl_t::fini()
   return RMW_RET_OK;
 }
 
+#ifdef DDS_HAS_SHM
+template<typename entityT>
+static void * init_and_alloc_sample(
+  entityT & entity, const uint32_t sample_size, const bool alloc_on_heap = false)
+{
+  // initialise the data allocator
+  if (alloc_on_heap) {
+    RET_EXPECTED(
+      dds_data_allocator_init_heap(&entity->data_allocator),
+      DDS_RETCODE_OK,
+      "Reader data allocator initialization failed for heap",
+      return nullptr);
+  } else {
+    RET_EXPECTED(
+      dds_data_allocator_init(entity->enth, &entity->data_allocator),
+      DDS_RETCODE_OK,
+      "Writer allocator initialisation failed",
+      return nullptr);
+  }
+  // allocate memory for message + header
+  auto chunk_ptr = dds_data_allocator_alloc(
+    &entity->data_allocator,
+    DETERMINE_ICEORYX_CHUNK_SIZE(sample_size));
+  RMW_CHECK_FOR_NULL_WITH_MSG(
+    chunk_ptr,
+    "Failed to get loan",
+    return nullptr);
+  auto ice_hdr = static_cast<iceoryx_header_t *>(chunk_ptr);
+  ice_hdr->data_size = sample_size;
+  auto ptr = SHIFT_PAST_ICEORYX_HEADER(chunk_ptr);
+  // initialize the memory for message
+  rmw_cyclonedds_cpp::init_message(&entity->type_supports, ptr);
+  return ptr;
+}
+
+template<typename entityT>
+static rmw_ret_t fini_and_free_sample(entityT & entity, void * loaned_message)
+{
+  // fini the message
+  rmw_cyclonedds_cpp::fini_message(&entity->type_supports, loaned_message);
+  // free the message memory
+  RET_EXPECTED(
+    dds_data_allocator_free(
```

Comment:
IIUC `0.8.0beta5` doesn't support this function, right?

We're currently using that version in `galactic` https://github.com/ros2-gbp/cyclonedds-release/blob/92d3232d8f8461a87693b9b7aa285698794cec4e/tracks.yaml#L104, so we would need to first update it.

Reply:
You're right, that was already part of beta5.
CI will catch the issue if it wasn't, so I will just approve.